tests/layer_tests/tensorflow_tests/test_tf_Range.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+
+
+class TestRange(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        inputs_data = {}
+        if self.negative_delta:
+            inputs_data['start'] = np.random.randint(1, 10, []).astype(self.input_type)
+            inputs_data['limit'] = np.random.randint(-10, 0, []).astype(self.input_type)
+            inputs_data['delta'] = np.random.randint(-5, -1, []).astype(self.input_type)
+        else:
+            inputs_data['start'] = np.random.randint(1, 10, []).astype(self.input_type)
+            inputs_data['limit'] = np.random.randint(10, 30, []).astype(self.input_type)
+            inputs_data['delta'] = np.random.randint(1, 5, []).astype(self.input_type)
+
+        return inputs_data
+
+    def create_range_net(self, input_type, negative_delta):
+        self.input_type = input_type
+        self.negative_delta = negative_delta
+        types_map = {
+            np.float32: tf.float32,
+            np.int32: tf.int32
+        }
+        assert input_type in types_map, "Incorrect test case"
+        tf_type = types_map[input_type]
+        tf.compat.v1.reset_default_graph()
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            start = tf.compat.v1.placeholder(tf_type, [], 'start')
+            limit = tf.compat.v1.placeholder(tf_type, [], 'limit')
+            delta = tf.compat.v1.placeholder(tf_type, [], 'delta')
+            tf.raw_ops.Range(start=start, limit=limit, delta=delta)
```

Comment:
please try to set `dtype` attribute with different values to Range op here.
It looks like, it can be related with this deleted line
`auto out_type = node.get_attribute<ov::element::Type>("Tidx");`

Reply:
`tf.raw_ops.Range` does not have `dtype` attribute: https://www.tensorflow.org/api_docs/python/tf/raw_ops/Range
And according to the specification `Tidx` can be absent at all so we can't rely on it. It explicitly says that output type is the same as `start`

Reply:
but the python version has https://www.tensorflow.org/api_docs/python/tf/range
![image](https://user-images.githubusercontent.com/21181928/217197397-de031bd3-d5d9-4af8-8ae2-c9992fe8f834.png)
please check how it works

Reply:
I just checked. Everytime it finally converted to `raw_ops.Reshape` version that follows to the specification.
It is the same for all operations. So we need to look at `raw_ops`:)